### DISCUSSION TITLE
Sync buffer paths with renames

### DISCUSSION THREAD

**Discussion by onsails** [mcchrish/nnn.vim#117] [Ideas]
Suppose I have a buffer opened with path a/b/foo.bar
Then I open nnn and rename b to c
Then I modify buffer a/b/foo.bar, write it and get an error
```
E212: Can't open file for writing: no such file or directory
E211: File "a/b/foo.bar" no longer available
```
which makes perfect sense but in my opinion it would be more conveninent if nnn.vim could optionally change buffer path matching renaming entity to the new path according to the rename.

**Comment by mcchrish** [comment]
I think this need some request to upstream nnn to somehow publish events (either via a server or just a simple fifo just like how NnnExplorer works). Once the plugin understands and receive events from nnn, it can then react and change things in vim.
For example:
- Delete buffer when the file is deleted
- Rename buffer when file is rename
- Move...
- And so on.

A server mode for nnn would actually be a great feature that opens lots of possible use cases.
